sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -3714,12 +3769,13 @@ atomic_update(T *p, simd<Toffset, N> byte_offset, simd_mask<N> mask,
   } else {
     if constexpr (Op == atomic_op::load) {
       if constexpr (std::is_integral_v<T>) {
-        return atomic_update<atomic_op::bit_or, T, N>(p, byte_offset,
-                                                      simd<T, N>(0), mask);
+        return atomic_update<atomic_op::bit_or, T, N>(
+            p, byte_offset, simd<T, N>(0), mask, props);
```

Comment:
Being in this branch means that both L1H/L2H are cache_hint::none, and we expect DWORD atomic. We can still do this for consistency, as these APIs should also expected to produce non-lsc variants. It's true that we do not have LIT coverage for all opcodes, so these things may go unnoticed.

Reply:
yeah my idea was maybe we add some property in the future that we need in those functions so might as well pass it down

Reply:
Yes, sure, good idea.